lib/extendedkey.ex

```diff
@@ -72,25 +89,36 @@ defmodule Bitcoinex.ExtendedKey do
     end
 
     defp str_to_level(level) do
-      [num | tick] = String.split(level, ["'", "h"])
+      {num, is_hardened} =
+        case String.split(level, ["'", "h"]) do
+          [num] ->
+            {num, false}
+
+          [num, ""] ->
+            {num, true}
+        end
+
       nnum = String.to_integer(num)
 
-      if nnum >= @softcap or nnum < 0 do
-        raise(ArgumentError, message: "invalid derivation path")
+      if nnum in @min_non_hardened_child_num..@max_non_hardened_child_num do
+        if is_hardened do
+          nnum + @min_hardened_child_num
+        else
+          nnum
+        end
       else
-        harden(nnum, tick)
+        raise(ArgumentError, message: "invalid derivation path")
```

Comment:
We should convert this to {:error, "invalid derivation path"}. my bad

Reply:
This one take more effort as it's in child_num path level and raising error is the approach used in your branch.
I think it should be a separate PR if we decide to return error tuple for this case

Reply:
Great, I can do it in a commit.